### PR TITLE
Some random stuff I ran into

### PR DESCRIPTION
**Pull request by durandom**
Hey,
nice stuff!

WDYT of those small additions?
If you think :+1: I can make this a proper PR :)

### CODE REVIEW

File: lib/dug/runner.rb [hunk 1]
```diff
@@ -33,6 +33,7 @@ def unprocessed_notifications(use_cache: true)
         @unprocessed_notifications = nil
       end
       unprocessed_label = servicer.labels(use_cache: use_cache)[Dug.configuration.unprocessed_label_name]
+      raise "Create the label '#{Dug.configuration.unprocessed_label_name}' on gmail" unless unprocessed_label
```
Comment (chrisarcand) [inline lib/dug/runner.rb:36]: Huh, thought I remember doing this somewhere else... but good add.

File: lib/dug/configurator.rb [hunk 2]
```diff
@@ -9,6 +9,7 @@ class Configurator
     attr_accessor :token_store
     attr_accessor :application_credentials_file
     attr_accessor :unprocessed_label_name
+    attr_accessor :default_label
```
Comment (chrisarcand) [inline lib/dug/configurator.rb:12]: I guess a more descriptive name would just be `processed_label_name`, as it's simply the opposite of the unprocessed one.

File: lib/dug.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@
 
 module Dug
   LABEL_RULE_TYPES = %w(organization repository reason state)
-  GITHUB_REASONS = %w(author comment mention team_mention state_change assign manual subscribed)
+  GITHUB_REASONS = %w(author comment mention team_mention state_change assign manual subscribed push)
```
Comment (chrisarcand) [inline lib/dug.rb:13]: Is this true? I don't see `push` in the [API documentation](https://developer.github.com/v3/activity/notifications/#notification-reasons) _but_ I know that Github tweaks it's Notification API all the time and don't really change their docs, so I believe it.

**Comment by chrisarcand** [comment]
These additions are completely sane and welcome if you'd like to refine them, absolutely. ❤️

**Comment by chrisarcand** [comment]
`push` bug merged already in https://github.com/chrisarcand/dug/pull/25/commits/869292b178a4eaa0996b8b9f82d2db5ccba2537e since its a blocking bug.

File: lib/dug/runner.rb (hunk 1 above)
Comment (blomquisg) [inline lib/dug/runner.rb:36]: It's auth'd via OAuth, right?  Can it just create the label for you?  Or, maybe have that configurable in the yaml...

Comment (chrisarcand) [inline lib/dug/runner.rb:36]: Yes. I hadn't yet gotten to more user friendly 'auto create' stuff and didn't know how I felt about it. 

One thought for people who _cough_ might want to implement this for us _cough_: Dug should _not_ auto create labels for you using the main runner. If we want to have some sort of way that labels are created for you I'd like that as a separate action, something like `Dug.configure`, whatever. Really some bins to run the runner and and configure things for you would be nice.

Comment (chrisarcand) [inline lib/dug/runner.rb:36]: Forgot to say... The configure step would check your current .yml and ensure the labels you have in there actually exist in Gmail and create them for you if they don't (maybe prompt by prompt)

**Comment by durandom** [comment]
created smaller PRs

**Pull request closed without merging**